datahub-web-react/src/app/ingest/source/builder/RecipeForm/dbt_cloud.tsx

```diff
@@ -0,0 +1,296 @@
+    },
+};
+
+const nodeAllowFieldPath = 'source.config.node_name_pattern.allow';
+export const NODE_ALLOW: RecipeField = {
+    name: 'node_name_pattern.allow',
+    label: 'Allow Patterns',
+    tooltip:
+        'Only include specific dbt Nodes (resources) by providing their name, or a Regular Expression (REGEX). If not provided, all Nodes will be included.',
+    placeholder: 'model_name',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: nodeAllowFieldPath,
+    rules: null,
+    section: 'Nodes',
+};
+
+const nodeDenyFieldPath = 'source.config.node_name_pattern.deny';
+export const NODE_DENY: RecipeField = {
+    name: 'node_name_pattern.deny',
+    label: 'Deny Patterns',
+    tooltip:
+        'Exclude specific dbt Nodes (Resources) by providing their name, or a Regular Expression (REGEX). If not provided, all Nodes will be included. Deny patterns always take precedence over Allow patterns.',
+    placeholder: 'node_name',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: nodeDenyFieldPath,
+    rules: null,
+    section: 'Nodes',
+};
+
+export const METADATA_ENDPOINT: RecipeField = {
+    name: 'metadata_endpoint',
+    label: 'Custom Metadata Endpoint URL',
+    tooltip:
+        'A custom URL used for extracting Metadata. By default, this metadata is extracted from https://metadata.cloud.getdbt.com/graphql. In most cases, users should NOT need to provide this value.',
+    placeholder: 'https://metadata.cloud.getdbt.com/graphql',
+    type: FieldType.TEXT,
+    fieldPath: 'source.config.metadata_endpoint',
+    rules: null,
+};
+
+const extractOwnersPath = 'source.config.enable_owner_extraction';
+export const EXTRACT_OWNERS: RecipeField = {
+    name: 'extract_owners',
+    label: 'Extract Owners',
+    tooltip:
+        'Try to extract owners from dbt meta properties. Be careful: This can override Owners added by users of DataHub.',
+    type: FieldType.BOOLEAN,
+    fieldPath: 'source.config.enable_owner_extraction',
+    rules: null,
+    getValueFromRecipeOverride: (recipe: any) => {
+        const extractOwners = get(recipe, extractOwnersPath);
+        if (extractOwners !== undefined && extractOwners !== null) {
+            return extractOwners;
+        }
+        return true;
+    },
+};
+
+export const TARGET_PLATFORM: RecipeField = {
+    name: 'target_platform',
+    label: 'Data Platform (Connection Type)',
+    tooltip: 'The type of Data Platform that dbt is connected to.',
+    placeholder: 'Select a Data Platform Type...',
+    type: FieldType.SELECT,
+    options: [
+        { label: 'Snowflake', value: 'snowflake' },
+        { label: 'BigQuery', value: 'bigquery' },
+        { label: 'Redshift', value: 'redshift' },
+        { label: 'Postgres', value: 'postgres' },
+        { label: 'Trino (Starburst)', value: 'trino' },
+        { label: 'Databricks', value: 'databricks' },
+    ],
```

Comment:
nice - it's much better to have a select than a text box in the normal case.

Is there an escape hatch to allow ppl to enter in a custom string e.g. if they're using databricks, clickhouse, etc? dbt has a lot of community-supported adapters https://docs.getdbt.com/docs/supported-data-platforms

Reply:
The escape hatch right now wil be the YAML. We will not support switch to text box dynamically... yet :)

Reply:
want to need this for a couple cases before building the generalized infra for it